### PR TITLE
Supporting illegal whitespace characters in chunk size #1812

### PR DESCRIPTION
**Pull request by SakulK**

### CODE REVIEW

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala [hunk 1]
```diff
@@ -255,6 +255,16 @@ abstract class RequestParserSpec(mode: String, newLine: String) extends FreeSpec
...
+      "with incorrect but harmless whitespace after chunk size" in new Test {
+        Seq(
+          start,
+          """0  
+            |
+            |""") should generalMultiParseTo(
+            Right(baseRequest.withEntity(Chunked(`application/pdf`, source(LastChunk)))))
+        closeAfterResponseCompletion shouldEqual Seq(false)
+      }
+
```
Comment (ktoso) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala]: thanks for the test 👍

File: akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala [hunk 2]
```diff
@@ -243,6 +243,7 @@ private[http] trait HttpMessageParser[Output >: MessageOutput <: ParserOutput] {
           case ';' if cursor > offset ⇒ parseChunkExtensions(size.toInt, cursor + 1)()
           case '\r' if cursor > offset && byteChar(input, cursor + 1) == '\n' ⇒ parseChunkBody(size.toInt, "", cursor + 2)
           case '\n' if cursor > offset ⇒ parseChunkBody(size.toInt, "", cursor + 1)
+          case c if CharacterClasses.WSP(c) ⇒ parseSize(cursor + 1, size)
```
Comment (ktoso) [inline akka-http-core/src/main/scala/akka/http/impl/engine/parsing/HttpMessageParser.scala]: perhaps we should add some comment that we know this is illegal in the spec but we've seen it in the wild?

**Comment by raboof** [comment]
OK TO TEST

File: akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala [hunk 3]
```diff
@@ -255,6 +255,16 @@ abstract class RequestParserSpec(mode: String, newLine: String) extends FreeSpec
         closeAfterResponseCompletion shouldEqual Seq(false)
       }
 
+      "with incorrect but harmless whitespace after chunk size" in new Test {
+        Seq(
+          start,
+          """0
```
Comment (jrudolph) [inline akka-http-core/src/test/scala/akka/http/impl/engine/parsing/RequestParserSpec.scala]: Maybe encode the whitespace as `\u0020` so it's clearer.

**Comment by jrudolph** [comment]
> The worst thing that could happen is the method spinning for a while so that's not nice

It's not really spinning, it's still parsing and making progress. The method itself is `@tailrec` so that's fine.

> If we're worried about that aspect we should hide this under a relaxed mode flag.

I'd say it's fine like this

**Comment by ktoso** [comment]
Merge once validated

**Comment by ktoso** [comment]
Damn those formatting failures...! Thanks @jrudolph for fixing those.
Ready to go then.

Thanks for the fix @SakulK !

**Comment by ktoso** [comment]
Resolves #1812

**Comment by SakulK** [comment]
Thanks for your help @ktoso @jrudolph :)

**Pull request merged**